staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -4512,6 +4539,26 @@ func (c *Cloud) getFullInstance(nodeName types.NodeName) (*awsInstance, *ec2.Ins
 	return awsInstance, instance, err
 }
 
+func (c *Cloud) nodeNameToProviderID(nodeName types.NodeName) (InstanceID, error) {
+	if len(nodeName) == 0 {
+		return "", fmt.Errorf("no nodeName provided")
+	}
+
+	if c.nodeInformerHasSynced == nil || !c.nodeInformerHasSynced() {
+		return "", fmt.Errorf("node informer has not synced yet")
+	}
+
+	node, err := c.nodeInformer.Lister().Get(string(nodeName))
+	if err != nil {
```

Comment:
Does this require a RBAC policy change so as AWS cloudprovider can list nodes?

Reply:
Good question! I checked that after v1.15, we have deprecated self-provisioned cloud provider role, and have user to create them https://github.com/kubernetes/kubernetes/pull/66635

I will modify the release note about the face that we need "list" permission now

Reply:
Also, @justinsb we might want to modify kops in this case as well? https://github.com/kubernetes/kops/pull/6899

Reply:
I do not remember if we need a RBAC rule if we are using a shared informer. If we do not - we are probably fine.  We need to confirm that...

Reply:
actually @justinsb I did some tests and verified that we do not need additional RBAC rules for `system:aws-cloud-provider` to use shared informer. Our clusters are provisioned using kops and all roles contain default values